wal.go

```diff
@@ -523,6 +523,12 @@ func (w *SegmentWAL) openSegmentFile(name string) (*os.File, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	//Sync the file first to see any unflushed buffers.
+	if err := fileutil.Fdatasync(f); err != nil {
+		return nil, err
+	}
+
```

Comment:
I really believe this should[tm] not be necessary but doesn't seem like the worst idea for safety.

Space missing in front of comment.